### ISSUE TITLE
"Outline > simple line" does not grant access to the options/library of lines symbols

### ISSUE THREAD

**Issue by gioman** [qgis/QGIS#58377]
### What is the bug or the crash?

Fora polygon layer when you choose its style as "outline > simple line" QGIS does not grant access to the options/library for line synmbols, but if you choose "outline > marker line" then it correctly shows the options/library for point symbols

![image](https://github.com/user-attachments/assets/f64233e4-1a87-463b-97dd-32481901913f)

VS

![image](https://github.com/user-attachments/assets/9fe65268-ec25-4643-97bc-0b6a307a27dd)

### Steps to reproduce the issue

Add a polygon layer

Style it with "outline > simple line"

### Versions

3.34
3.38
master

<table width='100%' align='center'><tr><td colspan="2"><b>Libraries</b></td><tr><td>QGIS version</td><td>3.39.0-Master</td></tr><tr><td>QGIS code revision</td><td><a href="https://github.com/qgis/QGIS/commit/29bf082ed3">29bf082ed3</a></td></tr><tr><td>Qt version</td><td>5.15.13</td></tr><tr><td>Python version</td><td>3.12.5</td></tr><tr><td>GDAL/OGR version</td><td>3.10.0dev-6339b46c60</td></tr><tr><td>PROJ version</td><td>9.4.0</td></tr><tr><td>EPSG Registry database version</td><td>v11.004 (2024-02-24)</td></tr><tr><td>GEOS version</td><td>3.12.2-CAPI-1.18.2</td></tr><tr><td>SQLite version</td><td>3.45.1</td></tr><tr><td>PDAL version</td><td>2.6.3</td></tr><tr><td>PostgreSQL client version</td><td>16.2</td></tr><tr><td>SpatiaLite version</td><td>5.1.0</td></td></tr><tr><td>QWT version</td><td>6.2.0</td></tr><tr><td>QScintilla2 version</td><td>2.14.1</td></tr><tr><td>OS version</td><td>Windows 11 Version 2009</td></tr><tr><td colspan="2"><i>This copy of QGIS writes debugging output.</i></td></tr><tr><td colspan="2">&nbsp;</td></tr><tr><td colspan="2"><b>Active Python plugins</b></td></tr><tr><td>fccn_validator</td><td>1.0.0</td></tr><tr><td>lizmap</td><td>4.3.19</td></tr><tr><td>openlittermap</td><td>1.0.0</td></tr><tr><td>db_manager</td><td>0.1.20</td></tr><tr><td>processing</td><td>2.12.99</td></tr></table>

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by nyalldawson** [comment]
This isn't a bug -- the symbol library pane is only shown when you are configuring a sub-symbol for a symbol layer, so it's only visible for symbol layer types which use sub symbols. Marker line uses these, simple line does not.

**Comment by gioman** [comment]
> This isn't a bug -- the symbol library pane is only shown when you are configuring a sub-symbol for a symbol layer, so it's only visible for symbol layer types which use sub symbols. Marker line uses these, simple line does not.

@nyalldawson ummm... I see... but then what about if I want to style the outline of a polygon with a line style I previously saved in my library? Isn't possible? Seems weird (from a user point of view) considered all we can do with symbology in QGIS.

**Comment by nyalldawson** [comment]
@gioman yeah, that's a good point. I think if we want to handle this then we could expose line symbols in the top-level browser for polygon layers too.
